test/catalog_test.exs

```diff
@@ -34,77 +35,87 @@ defmodule Wanda.CatalogTest do
+                remediation: "## Remediation\nRemediation text\n",
+                severity: :critical,
+                facts: [
+                  %Fact{
+                    name: "jedi",
+                    gatherer: "wandalorian",
+                    argument: "-o"
+                  },
+                  %Fact{
+                    name: "other_fact",
+                    gatherer: "no_args_gatherer",
+                    argument: ""
+                  }
+                ],
+                values: [
+                  %Value{
+                    conditions: [
+                      %Condition{expression: "some_expression", value: 10},
+                      %Condition{expression: "some_other_expression", value: 15}
+                    ],
+                    default: 5,
+                    name: "expected_value"
+                  },
+                  %Value{
+                    conditions: [
+                      %Condition{expression: "some_third_expression", value: 5}
+                    ],
+                    default: 10,
+                    name: "expected_higher_value"
+                  }
+                ],
+                expectations: [
+                  %Expectation{
+                    name: "some_expectation",
+                    type: :expect,
+                    expression: "facts.jedi == values.expected_value"
+                  },
+                  %Expectation{
+                    name: "some_other_expectation",
+                    type: :expect,
+                    expression: "facts.jedi > values.expected_higher_value"
+                  }
+                ]
+              }} = Catalog.get_check("expect_check")
     end
 
     test "should load a expect_same expectation type" do
-      assert %Check{
-               values: [],
-               expectations: [
-                 %Expectation{
-                   name: "some_expectation",
-                   type: :expect_same,
-                   expression: "facts.jedi"
-                 }
-               ]
-             } = Catalog.get_check("expect_same_check")
+      assert {:ok,
+              %Check{
+                values: [],
+                expectations: [
+                  %Expectation{
+                    name: "some_expectation",
+                    type: :expect_same,
+                    expression: "facts.jedi"
+                  }
+                ]
+              }} = Catalog.get_check("expect_same_check")
     end
 
     test "should load a warning severity" do
-      assert %Check{severity: :warning} = Catalog.get_check("warning_severity_check")
+      assert {:ok, %Check{severity: :warning}} = Catalog.get_check("warning_severity_check")
+    end
+
+    test "should return an error for non-existent check" do
+      assert {:error, _} = Catalog.get_check("non_existent_check")
+    end
+
+    test "should return an error for malformed check" do
+      assert {:error, _} = Catalog.get_check("malformed_check")
```

Comment:
Let's pattern match here the expected error, which is `{:error, :malformed_check}`